arviz/plots/backends/bokeh/densityplot.py

```diff
@@ -91,6 +90,17 @@ def plot_density(
                 axis_map[label],
                 data_label=data_label,
             )
+            if data_label is not None:
+                if axis_map[label] in legend_items.keys():
+                    legend_items[axis_map[label]].append((data_label, plotted))
+                else:
+                    legend_items[axis_map[label]] = []
+                    legend_items[axis_map[label]].append((data_label, plotted))
+
+    for ax1, lg in legend_items.items():
+        legend = Legend(items=lg, location="center_right", orientation="horizontal",)
```

Comment:
I think this is a good compromise.

I don't think there is any "universal" way to position legend so it looks good